File: library/src/main/java/com/mikepenz/fastadapter/commons/utils/DiffCallbackImpl.java
package com.mikepenz.fastadapter.commons.utils;

import android.support.annotation.Nullable;

import com.mikepenz.fastadapter.IItem;

/**
 * Created by mikepenz on 24.08.16.
 */

public class DiffCallbackImpl<Item extends IItem> implements DiffCallback<Item> {
    @Override
    public boolean areItemsTheSame(Item oldItem, Item newItem) {
        return oldItem.getIdentifier() == newItem.getIdentifier();
    }

    @Override
    public boolean areContentsTheSame(Item oldItem, Item newItem) {
        return oldItem.equals(newItem);
    }

    @Nullable
    @Override
    public Object getChangePayload(Item oldItem, int oldItemPosition, Item newItem, int newItemPosition) {
        return null;
    }
}

File: library-extensions-diff/src/main/java/com/mikepenz/fastadapter/diff/DiffCallbackImpl.kt
package com.mikepenz.fastadapter.diff

import com.mikepenz.fastadapter.IItem

/**
 * Created by mikepenz on 24.08.16.
 */
internal class DiffCallbackImpl<Item : IItem<*>> : DiffCallback<Item> {
    override fun areItemsTheSame(oldItem: Item, newItem: Item): Boolean {
        return oldItem.identifier == newItem.identifier
    }

    override fun areContentsTheSame(oldItem: Item, newItem: Item): Boolean {
        return oldItem == newItem
    }

    override fun getChangePayload(oldItem: Item, oldItemPosition: Int, newItem: Item, newItemPosition: Int): Any? {
        return null
    }
}
